composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/data/TaskyDatabase.kt

```diff
@@ -17,6 +23,18 @@ import sp.bvantur.tasky.core.data.local.EventEntity
 abstract class TaskyDatabase : RoomDatabase() {
     abstract fun getAttendeeDao(): AttendeeDao
     abstract fun getEventDao(): EventDao
+
+    suspend fun clearDatabase(): TaskyResult<Unit, TaskyError> = try {
+        useWriterConnection {
+            it.immediateTransaction {
+                getAttendeeDao().removeAllAttendeeData()
+                getEventDao().removeAllEventData()
```

Comment:
How would you guys handle this clearing of the data? Is LocalDataSource a better fit, it is ok to have such a method written directly in Database object?

Reply:
I tend to leave the clearing of tables inside their respective DAOs which I would call when logging the user out.

```
private fun logout() {
  applicationScope.launch {
    syncAgendaItemsScheduler.cancelAllSyncs()
    agendaRepository.deleteAllAgendaItems()
    authCoreService.logout()
    sessionStorage.setSession(null)
  }
}
```

Inside my local data source I have the following that clears each DAO

```
override suspend fun deleteAllAgendaItems() {
  database.withTransaction {
    withContext(dispatcherProvider.io) {
      listOf(
        async { eventDao.deleteAllEvents() },
        async { reminderDao.deleteAllReminders() },
        async { taskDao.deleteAllTasks() }
      ).awaitAll()
    }
  }
}
```